### ISSUE TITLE
Link styles: consider using the default focus outline style set by the browsers

### ISSUE THREAD

**Issue by marctison75** [WordPress/twentytwentyone#444]
Following Jeff Ong’s suggestion to create a new issue, I’m opening this one based on the discussion of  #165.

The current link styles don’t look good on Safari. 

![Safari](https://user-images.githubusercontent.com/72606415/95842202-8c70b300-0d46-11eb-903d-0cf82c28bc15.png)

In Firefox on Windows, you can see how the outline can interact with the shape of the letter and decrease the readability of the text.

![Firefox](https://user-images.githubusercontent.com/72606415/95842250-9a263880-0d46-11eb-96d8-3c3484a3a52d.jpg)

A solution might be to use the default focus style set by the browsers (user agent stylesheet), which is already an outline (https://www.w3.org/ is using it). This solution is close to the initial design and doesn’t have any issues.

**Comment by sarahricker** [comment]
Thanks for pointing this out @marcduluc!

@melchoyce a11y and Design discussed this one in our new weekly office hours. Overall, we don’t think the browser’s default outline style is enough. We can definitively design something better. It seems though, that this issue might be caused by the larger border width, possibly also line-height and the wrapping concern. 

Am I right this was intended to match the other borders on the site (like the search box)? Asking, because keyboard focus on the search box should also have a focus state, but it seems the input box is getting double boarder while the button gets an inset dashed line.

Seems we need a more global focus style that would better fit all concerns, but I'm open to ideas!

**Comment by carolinan** [comment]
There is not a lot of time left to redesign the links again. Please be mindful that it is late into the project now.
For example the last call for features was yesterday.

**Comment by sarahricker** [comment]
Gotcha, thanks @carolinan! Maybe we can limit the focus here to the cross-browser bugs.

**Comment by enriquesanchez** [comment]
I think we can try some small variations that should improve the focus style, I don't think these to be big changes but I'll leave it to the project leads to consider.

**Use the default browser focus style**

![2020-10-14 10-02-28 2020-10-14 10_03_04](https://user-images.githubusercontent.com/3276087/96008486-3adb2d80-0e05-11eb-8b46-c1fdd0ed1f40.gif)

**Change the thickness of the underline**

![2020-10-14 09-59-56 2020-10-14 10_00_51](https://user-images.githubusercontent.com/3276087/96008571-50e8ee00-0e05-11eb-9879-b1388d65f6b8.gif)

**Use a background color**

![2020-10-14 10-06-01 2020-10-14 10_06_27](https://user-images.githubusercontent.com/3276087/96008647-63fbbe00-0e05-11eb-9f82-25b5af0f059f.gif)

**A combination of underline and background color**

![2020-10-14 10-00-59 2020-10-14 10_02_11](https://user-images.githubusercontent.com/3276087/96008692-7118ad00-0e05-11eb-8bc2-5b739ab4736e.gif)

If we consider that the goal of the focus style is to make it easy to spot which element is in focus, I think it's OK if we go with a style that is prominent.

That said, if it's already too late in the development process, then I'd suggest we stick with the browser defaults.

**Comment by melchoyce** [comment]
Let's try browser defaults for all focus styles. Can someone make a PR?

**Comment by carolinan** [comment]
The default outline style still overlap in Firefox;

![overlap-outline](https://user-images.githubusercontent.com/7422055/96144483-832d4500-0f04-11eb-8535-20db58be9c13.png)

![overlap-outline2](https://user-images.githubusercontent.com/7422055/96144617-a821b800-0f04-11eb-90a6-9457eb84d77b.png)

And the style is almost invisible:
![outline-black](https://user-images.githubusercontent.com/7422055/96144789-d7d0c000-0f04-11eb-95e8-66dc867adf7d.png)
![outline](https://user-images.githubusercontent.com/7422055/96144791-d8695680-0f04-11eb-8152-202c57ed9fcf.png)

**Comment by sarahricker** [comment]
Discussed again in a11y + Design office hours (see [slack conversation](https://wordpress.slack.com/archives/C02RP4X03/p1603206930321200)).

Those present discussed the different possibilities, and generally agreed that their preference leaned towards a combination of underline and background color, much like the [W3's own website](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html). 

The only issue with that approach is choosing an appropriate background color to retain a 4.5:1 color contrast for multiple color schemes, as @sunnychan pointed out:
> The challenge is we cannot make general assumptions what the background colour behind the link is. So if a user add a block with a white background, unless you are able to select the contrast colour for the focus state, you will not see the contrast

However, @enriquesanchez noted:
>I think this is also important to note that the theme will be WCAG AAA ready. Because we can ensure that all defaults meet guidelines, but that still means the author can change any value/setting and break compliance.

Given that today is Beta 1, I don't think much more will make it in. However, I suggest keep this issue open for possible future theme updates. 

Thank you to everyone working to untangle this one!

**Comment by melchoyce** [comment]
With a white background color applied, light scheme:

![image](https://user-images.githubusercontent.com/2846578/96619686-788ff880-12d4-11eb-99ba-d2b8d2e90e66.png)

With a dark background color applied, dark scheme:

![image](https://user-images.githubusercontent.com/2846578/96619792-965d5d80-12d4-11eb-8099-28f4d1b8e1ac.png)

Is this what you're picturing @sarahricker / @enriquesanchez? 

They get a little weird, since the background color doesn't appear within the space between lines, so there's a gap.

**Comment by enriquesanchez** [comment]
>Is this what you're picturing

@melchoyce Almost. I agree it looks weird with the space between lines. What if it was bg color plus underline? Something like the 3rd example in [my comment above](https://github.com/WordPress/twentytwentyone/issues/444#issuecomment-708468634).

We already have the underline for links, we'd just be adding the bg color.

**Comment by sarahricker** [comment]
Yep @melchoyce ! That looks lovely too me. If @enriquesanchez's border suggestion fixes the gap, that should be prefect for most common colors!

Is this fix working well cross-browser as well? I can help test tomorrow.

**Comment by afercia** [comment]
During yesterday's accessibility office-hours on Slack, @enriquesanchez suggested to have a look at https://www.gov.uk/ and https://www.a11yproject.com/ for inspiration and I'd totally second that suggestion.

Also, quoting from https://github.com/WordPress/twentytwentyone/pull/507#pullrequestreview-510361371
> native focus styles ... give the advantage they can easily be overridden by users who use custom stylesheets for their special needs and by assistive technology tools that augment the focus styles.

However:

> From a theme that aims to be level AAA "ready", I would expect something better than the native focus styles. Websites that are strongly focused on accessibility do use very strong focus styles. One example for all: https://www.gov.uk/

Seems to me the background change proposed by @melchoyce goes in the right direction ❤️ Regarding gov.uk, worth noting the fact they do _not_ use a white background is intentional. White/black can have a too strong contrast for some users. Not saying the theme should use a yellow background like gov.uk, but maybe worth considering to avoid a full white.

**Comment by hedgefield** [comment]
This is looking good! I'm generally a fan of a splash of subtle color too, but I understand that that's not possible this late in the game, and I like the monochrome look too. So reading @afercia's suggestion, if we soften up the white background so the contrast with the text is softer (but still AAA), that would make everyone happy, right? The dark version already looks good to me.

The gap in background color is a bit annoying yeah, but I don't hate it. At first glance it kinda looks like it's an intentional design detail. Maybe we just roll with it? 😛

**Comment by melchoyce** [comment]
I'll mock up some ideas using a reduced opacity white!

**Comment by melchoyce** [comment]
Prototyped in CodePen: https://codepen.io/melchoyce/pen/QWEGEEW

I'm more drawn to the bottom suggestion from @enriquesanchez, now that I've tried both out. Would someone be able to try making a PR so we can test it out live?

**Comment by enriquesanchez** [comment]
@melchoyce The second style in the CodePen looks perfect to me 👍.

<img width="576" alt="Screen Shot 2020-10-22 at 16 53 46" src="https://user-images.githubusercontent.com/3276087/96934043-2c38ea00-1487-11eb-9856-3d165727daf0.png">

**Comment by rvpatel** [comment]
![image](https://user-images.githubusercontent.com/5517932/97113482-b8117700-1710-11eb-97e7-ba7e9b038669.png)
[https://files.slack.com/files-pri/T024MFP4J-F01DG1SNHA8/image.png](url)

Focus issue on gallery. This same open bug so not create new.
